tests/go-testcases/list.sky

```diff
@@ -214,36 +214,39 @@ assert_eq(bananas[4::-2], list("nnb".split_codepoints()))
 assert_eq(bananas[99::-2], list("snnb".split_codepoints()))
 assert_eq(bananas[100::-2], list("snnb".split_codepoints()))
 # TODO(adonovan): many more tests
-
+---
 # iterator invalidation
 def iterator1():
   list = [0, 1, 2]
   for x in list:
     list[x] = 2 * x
   return list
-assert_eq(iterator1(), [0, 2, 4]) # element updates are allowed
-
+# This is allowed in the go implementation but following the specification
```

Comment:
I don't understand what this means.  What is "this"? What is "they"? What is "deep content"? What is "them"?

What is the purpose of an assert_eq statement on the same line as a `###` comment indicating an expectation of failure?

Reply:
> I don't understand what this means. What is "this"? What is "they"? What is "deep content"? What is "them"?
"This" is the current statement, "They" the elements, "deep content" is defined in the spec linked in the comment. "Them" refers to they. 

Rephrased to make all that clearer.

> 
> What is the purpose of an assert_eq statement on the same line as a `###` comment indicating an expectation of failure?

I kept it to make the diff clearer but in the final implement it is indeed useless, removed.